src/python/pants/rules/core/test.py

```diff
@@ -125,14 +126,30 @@ def is_testable(
     return is_valid_target_type and has_sources
 
 
+class AddressAndTestResults(Collection[AddressAndTestResult]):
+  pass
+
+
 @dataclass(frozen=True)
 class AddressAndDebugRequest:
   address: BuildFileAddress
   request: TestDebugRequest
 
 
+@dataclass(frozen=True)
+class PytestCoverageReport:
+  digest: Digest
+  output_path: PurePath
```

Comment:
Did you decide what this should represent? The path to the file report or the path to the parent directory? That is, given the file report that should be saved at `coverage/python/html/report.html`, is this `coverage/python/html/` or `coverage/python/html/report.html`? If the former, consider renaming this to `output_directory`.